File: go/0071.0_Simplify_Path.go
/*
执行用时：4 ms, 在所有 Go 提交中击败了27.47% 的用户
内存消耗：2.6 MB, 在所有 Go 提交中击败了96.14% 的用户
通过测试用例：256 / 256
*/
func simplifyPath(path string) string {
    return filepath.Clean(path)
}

File: leetcode-cn/0071.0_Simplify_Path.py
'''
stack

执行用时：36 ms, 在所有 Python3 提交中击败了53.79% 的用户
内存消耗：15 MB, 在所有 Python3 提交中击败了69.10% 的用户
通过测试用例：256 / 256
'''
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for part in path.split('/'):
            if part == '..' and stack:
                stack.pop()
            elif part not in ('', '.', '..'):
                stack.append(part)

        return '/' + '/'.join(stack)

